File: Data Structures & Algorithms/reverse-string/submission-2.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        char first = 0;
        char last = s.size()-1;

        while (first < last) {
            char temp = s[first];
            s[first] = s[last];
            s[last] = temp;

            ++first;
            --last;
        }
    }
};

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-2.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        '''
        use heap
        - first element is closest
        - every element is farther
        '''
        max_heap = []
        for x, y in points:
            dist = x**2 + y**2
            if len(max_heap) < k:
                heapq.heappush(max_heap, (-dist, [x, y]))
            elif dist < -max_heap[0][0]:
                heapq.heapreplace(max_heap, (-dist, [x, y]))

        return [point for _, point in max_heap]

        